bin/sozo/src/ops/events.rs

```
@@ -40,124 +39,437 @@ pub async fn execute(
+                }
+                _ => Ok(format!("\"{}\"", hex)),
+            }
+        }
+        "bool" => {
+            if *value == FieldElement::ZERO {
+                Ok("false".to_string())
+            } else {
+                Ok("true".to_string())
+            }
+        }
+        "ClassHash" | "ContractAddress" => Ok(format!("{:#x}", value)),
+        "u8" | "u16" | "u32" | "u64" | "u128" | "usize" | "i8" | "i16" | "i32" | "i64" | "i128" => {
+            Ok(value.to_string())
+        }
+        _ => Err(format!("Unsupported CoreBasic type: {}", cb.type_name())),
+    }
+}
+
 fn parse_event(
     event: starknet::core::types::EmittedEvent,
-    events_map: &HashMap<String, Vec<Event>>,
+    events_map: &HashMap<String, Vec<Token>>,
 ) -> Option<String> {
-    let keys = event.keys;
-    let event_hash = keys[0].to_string();
-    let events = events_map.get(&event_hash)?;
+    let mut data = VecDeque::from(event.data.clone());
+    let mut keys = VecDeque::from(event.keys.clone());
+    let event_hash = keys.pop_front()?;
+
+    let events = events_map.get(&event_hash.to_string())?;
 
     'outer: for e in events {
-        let mut ret = format!("Event name: {}\n", e.name);
-        let mut data = VecDeque::from(event.data.clone());
+        let mut ret = format!("Event name: {}\n", e.type_path());
 
-        // Length is two only when its custom event
-        if keys.len() == 2 {
-            let name = parse_cairo_short_string(&keys[1]).ok()?;
-            ret.push_str(&format!("Model name: {}\n", name));
-        }
+        if let Token::Composite(composite) = e {
+            for inner in &composite.inners {
+                let result: Result<_, &'static str> = match inner.kind {
+                    CompositeInnerKind::Data => data.pop_front().ok_or("Missing data value"),
+                    CompositeInnerKind::Key => keys.pop_front().ok_or("Missing key value"),
+                    _ => Err("Unsupported inner kind encountered"),
```

Bcs the err is ignored later, it makes this error type pointless.